deps/media-playback/media-playback/media.c

```diff
@@ -653,93 +637,349 @@ static bool init_avformat(mp_media_t *m)
+		m->mp_state = mstate_paused;
+	return true;
+}
+
+/*Execute when media moves from paused to playing.*/
+static inline bool mp_media_resume(mp_media_t *m)
+{
+	if (m->mp_state == mstate_paused) {
+		m->mp_state = mstate_resume;
+
+		mp_media_push_message(m, mstate_playing);
 	}
-	if (!mp_media_reset(m)) {
+	return true;
+}
+
+static inline bool mp_media_seeking(mp_media_t *m)
+{
+	enum mp_media_state oldstate = m->mp_state;
+
+	m->mp_state = mstate_seeking;
+	int64_t seek_pos;
+
+	seek_pos = m->seek_pos;
+	m->seek_next_ts = true;
+
+	seek_to(m, seek_pos);
+
+	if (!m->eof) {
+		switch (oldstate) {
+		case mstate_paused:
+			mp_media_push_message(m, mstate_paused);
+			break;
+		default:
+			mp_media_push_message(m, mstate_playing);
+			break;
+		}
+	}
+	return true;
+}
+
+/*Executes when stop event occurs. One time action,
+  then enter the stopped state. */
+static inline bool mp_media_stopping(mp_media_t *m)
+{
+	m->mp_state = mstate_stopping;
+
+	mp_media_push_message(m, mstate_stopped);
+
+	return true;
+}
+
+/*Occurs after stopping but before another state*/
+static inline bool mp_media_stopped(mp_media_t *m)
+{
+	m->mp_state = mstate_stopped;
+
+	if (m->stop_cb)
+		m->stop_cb(m->opaque);
+
+	return true;
+}
+
+/*Flip the looping flag*/
+static inline bool mp_media_setloop(mp_media_t *m)
+{
+	enum mp_media_state oldstate = m->mp_state;
+
+	m->mp_state = mstate_setloop;
+
+	m->looping = !m->looping;
+
+	m->mp_state = oldstate;
+
+	return true;
+}
+
+/*Flip the reconnect flag*/
+static inline bool mp_media_setreconnect(mp_media_t *m)
+{
+	enum mp_media_state oldstate = m->mp_state;
+
+	m->mp_state = mstate_setreconnect;
+
+	m->reconnecting = !m->reconnecting;
+
+	m->mp_state = oldstate;
+
+	return true;
+}
+
+/*Executes when reset event occurs.*/
+static inline bool mp_media_resetting(mp_media_t *m)
+{
+	enum mp_media_state oldstate = m->mp_state;
+	m->mp_state = mstate_resetting;
+
+	if (!mp_media_reset(m, oldstate))
 		return false;
+
+	mp_media_push_message(m, mstate_playing);
+
+	return true;
+}
+
+/*Executes when kill event occurs.*/
+static inline bool mp_media_killing(mp_media_t *m)
+{
+	m->mp_state = mstate_killing;
+	m->killing = true;
+	return true;
+}
+
+/*Render the frame.*/
+static inline bool mp_media_rendering(mp_media_t *m, bool timeout)
+{
+	if (!timeout && m->mp_state != mstate_stopped) {
+		mp_render_frame(m);
 	}
 
-	for (;;) {
-		bool reset, kill, is_active, seek, pause, reset_time;
-		int64_t seek_pos;
-		bool timeout = false;
+	return true;
+}
 
-		pthread_mutex_lock(&m->mutex);
-		is_active = m->active;
-		pthread_mutex_unlock(&m->mutex);
+/*Executes after leaving a state. Keep in mind that this won't execute
+  until AFTER the entry method of the new state. Therefore, if cleanup
+  is needed before changing states, put that logic in the entry
+  function of the new state.*/
+static inline void mp_media_change_state(mp_media_t *m,
+					 enum mp_media_state oldstate)
+{
+	switch (oldstate) {
+	case mstate_uninit:
 
```

Comment:
If restructuring this into a switch statement, remove the blank lines between the `case` and `break` statements.